components/region_cache_memory_engine/src/engine.rs

```diff
@@ -23,19 +35,86 @@ pub struct RegionMemoryEngine {
     data: [Arc<Skiplist<ByteWiseComparator>>; 3],
 }
 
+impl RegionMemoryEngine {
+    pub fn with_capacity(arena_size: usize) -> Self {
+        RegionMemoryEngine {
+            data: [
+                Arc::new(Skiplist::with_capacity(
+                    ByteWiseComparator::default(),
+                    arena_size,
+                    true,
+                )),
+                Arc::new(Skiplist::with_capacity(
+                    ByteWiseComparator::default(),
+                    arena_size,
+                    true,
+                )),
+                Arc::new(Skiplist::with_capacity(
+                    ByteWiseComparator::default(),
+                    arena_size,
+                    true,
+                )),
+            ],
+        }
+    }
+}
+
+impl Default for RegionMemoryEngine {
+    fn default() -> Self {
+        RegionMemoryEngine {
+            data: [
+                Arc::new(Skiplist::with_capacity(
+                    ByteWiseComparator::default(),
+                    1 << 20,
+                    true,
+                )),
+                Arc::new(Skiplist::with_capacity(
+                    ByteWiseComparator::default(),
+                    1 << 20,
+                    true,
+                )),
+                Arc::new(Skiplist::with_capacity(
+                    ByteWiseComparator::default(),
+                    1 << 20,
+                    true,
+                )),
+            ],
+        }
+    }
+}
+
 impl Debug for RegionMemoryEngine {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        unimplemented!()
+        write!(f, "Region Memory Engine")
     }
 }
 
-type SnapshotList = BTreeMap<u64, u64>;
+#[derive(Default)]
+struct SnapshotList(BTreeMap<u64, u64>);
```

Comment:
<region_id, read_ts> may not be enough, we need ref count for this pair. Think about two snapshots with same region_id,read_ts.